packages/contract/src/test-utils/launch-test-node.ts

```diff
@@ -38,10 +40,20 @@ export interface LaunchTestNodeOptions<TContractConfigs extends DeployContractCo
    */
   contractsConfigs: TContractConfigs;
 }
+
+type ExtractDeployedContract<
+  T extends DeployContractConfig,
+  DeployMethod extends DeployableContractFactory['deploy'] = T extends DeployableContractFactory
+    ? T['deploy']
+    : T extends {
+          factory: DeployableContractFactory;
+        }
+      ? T['factory']['deploy']
+      : never,
+> = Awaited<ReturnType<Awaited<ReturnType<DeployMethod>>['waitForResult']>>['contract'];
+
```

Comment:
Who needs Cirque du Soleil when you've got TypeScript type juggling like this? 🤹‍♂️🤸‍♀️🎪 🤣

Reply:
Far from ideal.